include/signalrclient/hub_connection.h

```diff
@@ -28,6 +28,10 @@ namespace signalr
 
         SIGNALRCLIENT_API ~hub_connection();
 
+        hub_connection(const hub_connection&) = delete;

```

Comment:
This needs to be discussed when we review the API. Prevents from cycles by capturing the connection in the `hub_proxy.on()` callback but requires passing by reference or using a pointer. Note that this is not as bad as in the case of the `connection` since probably there is no reason to capture `hub_connection` in the `hub_proxy.on`. Capturing a `hub_proxy` instance is fine since it internally uses `weak_ptr` so there is no cycle.